### PR TITLE
boost-cpp v1.79.0

### PR DESCRIPTION
**Pull request by sdebionne**
Revert 86966baaba64e32d7a633fec09005afb777268e6 causing patch conflicts and already merged in the codebase.

### CODE REVIEW

**Comment by sdebionne** [comment]
@jakirkham This PR reverts the patch that was necessary for 1.78, could you merge it please?

**Comment by sdebionne** [comment]
Wait there is a crashing bug reported!

> These are patches from library authors which were found too late to be fixed in the release.
> 
> Boost.JSON array::erase can segfault, see [#692](https://github.com/boostorg/json/issues/692). [Patch](https://www.boost.org/patches/1_79_0/0001-json-array-erase-relocate.patch).

I am going to add it here.

- [ ] Add `0001-json-array-erase-relocate.patch`

**Comment by sdebionne** [comment]
There is another patch for Boost.Outcome that should be published soon.

**Comment by tschoonj** [comment]
> There is another patch for Boost.Outcome that should be published soon.

Is [this](https://github.com/ned14/outcome/pull/262) the patch you referred to?

**Comment by sdebionne** [comment]
Here is the patch that I was refereeing too:

https://lists.boost.org/Archives/boost/2022/04/252743.php

It's not clear to me why it's not published on the website though. I am adding the patch if it's OK for you.

**Comment by tschoonj** [comment]
Sure. I would just like to see this PR merged in soon.

File: recipe/0002-outcome-account-for-void-types.patch [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+@@ -22,6 +22,6 @@ Distributed under the Boost Software License, Version 1.0.
+ */
+ 
+ // Note the second line of this file must ALWAYS be the git SHA, third line ALWAYS the git SHA update time
+-#define OUTCOME_PREVIOUS_COMMIT_REF 147ec1e8673c34cb7cf431dfdbf211d8072d7656
+-#define OUTCOME_PREVIOUS_COMMIT_DATE "2022-03-17 19:38:05 +00:00"
+-#define OUTCOME_PREVIOUS_COMMIT_UNIQUE 147ec1e8
++#define OUTCOME_PREVIOUS_COMMIT_REF 5e36246352c1c412813583e446eea0d1a19d08cc
++#define OUTCOME_PREVIOUS_COMMIT_DATE "2022-04-05 06:23:15 +00:00"
++#define OUTCOME_PREVIOUS_COMMIT_UNIQUE 5e362463
```
Comment (jschueller) [inline recipe/0002-outcome-account-for-void-types.patch:14]: drop the revision changes

Comment (sdebionne) [reply recipe/0002-outcome-account-for-void-types.patch]: Revision changes are correct but the macro was renamed `BOOST_OUTCOME_ccc` in the meantime.

Comment (jschueller) [reply recipe/0002-outcome-account-for-void-types.patch]: check the git repo

**Pull request merged**